src/Form/SettingsForm.php

```diff
@@ -51,30 +43,26 @@ class SettingsForm extends ConfigFormBase {
    *   The config factory.
    * @param \Drupal\Core\Entity\EntityTypeManager $entityTypeManager
    *   The entity type manager.
-   * @param \Drupal\Core\Extension\ModuleHandlerInterface $moduleHandler
-   *   The module handler.
    * @param \Drupal\azure_ad_delta_sync\UserManagerInterface $userManager
    *   The user manager.
    *
    * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
    * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
    */
-  public function __construct(ConfigFactoryInterface $configFactory, EntityTypeManager $entityTypeManager, ModuleHandlerInterface $moduleHandler, UserManagerInterface $userManager) {
+  public function __construct(ConfigFactoryInterface $configFactory, EntityTypeManager $entityTypeManager, UserManagerInterface $userManager) {
     parent::__construct($configFactory);
     $this->userStorage = $entityTypeManager->getStorage('user');
     $this->roleStorage = $entityTypeManager->getStorage('user_role');
-    $this->moduleHandler = $moduleHandler;
     $this->userManager = $userManager;
   }
 
   /**
    * {@inheritdoc}
    */
-  public static function create(ContainerInterface $container) {
+  public static function create(ContainerInterface $container): SettingsForm {
```

Comment:
Consider using `static` as return type. 

```suggestion
  public static function create(ContainerInterface $container): static {
```

It doesn't change anything right nok, but it's easier to always write

```
  public static function create(ContainerInterface $container): static {
```

than

```
  public static function create(ContainerInterface $container): WhateverClassNameWereInside {
```

https://wiki.php.net/rfc/static_return_type